# Rabbit reproduction problem

<strong>Matthieu Nogatchewsky</strong><br>
Software & Machine Learning Engineer

<em>I am interested in data science in order to gain a wide array of new skills that will allow me to leverage data to aid companies with their business strategies, and explore exciting new fields developing from within data science—fields like artificial intelligence, machine learning, big data, and more.</em>

### The following code is supposed to calculate the number of pairs of rabbits in one year.

In [10]:
# Defining a function to compute the nth Fibonacci number.
def fibonacci(n):
    x_, x = 0, 1
    for i in range(2,n+1):
        x_, x = x, x_ + x
    return x

# Printing the number of pairs of rabbits at the 12th month given that in the first month there is only one pair of rabbits.
print(fibonacci(12))

144


This ~vital~ recreational problem was posed in the work [Liber abaci](https://en.wikipedia.org/wiki/Liber_Abaci).

> Someone put a pair of rabbits in a certain place, closed on all sides, to find out how many pairs would be born from this pair in one year, because it is in their nature to generate another pair in one month, and that they give birth in the second month after their birth.

Given the results printed in the above cell, the number of pairs of rabbits in one year is estimated at: <strong>144</strong>
